### PR TITLE
Use allowlist instead of blocklist in vscodeignore

### PR DESCRIPTION
**Pull request by t1m0thyj**
Remove extra files from the VSIX for both ZE and the FTP extension. Using an allowlist instead of a blocklist should make the `.vscodeignore` files easier to maintain in the future, as long as we remember to update the ignore list in any future PRs where new files should be included in the VSIX.

## Release Notes

Milestone: 2.6.0

Changelog: Not sure if we need one for this change? The only change users may notice is a smaller download size.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by zFernand0 — COMMENTED** [review]
I'm happy to see this PR.
Seems like this will definitely reduce the VSIX file size. 😋 

Should we document (in the docs folder perhaps) what things/new-features may require a change in the `.vscodeignore` files?

File: docs/README-Developer.md [hunk 1]
```diff
@@ -207,3 +208,14 @@ Maintainers: Evann Wu (evannw@andrew.cmu.edu), Lauren Li (lauren.li@ibm.com), Kr
    3. Affiliation with Zowe
 
 3. You should receive a response within 3 business days and be added to the Zanata Zowe VS Code Extension project. Click [here](http://docs.zanata.org/en/release/user-guide/translator-guide/) for more information about how to use Zanata to translate.
+
+## Packaging
+
+Run this command to create packages in the "dist" directory:
+`yarn workspace vscode-extension-for-zowe package`
```
Comment (JillieBeanSim) [inline docs/README-Developer.md:215]: should this line be `yarn package` to get all 3 packaged and in the dist dir.  Having `workspace vscode-extension-for-zowe` will only package the single vsix

**Review by JillieBeanSim — COMMENTED** [review]
Other than the single comment left, I have one more question.  How often would a developer need to change the .vscodeignore file?  I wouldn't think very often if just working within the project.

The size drop is awesome, I am seeing ZE from 2.1 MB => 770 KB and the FTP ext from 1.4 MB => 818 KB.  Thanks for this work @t1m0thyj

**Review by zFernand0 — DISMISSED** [review]
LGTM! 😋 
just waiting on the merge conflicts 
and maybe the readme update 😉

**Review by zFernand0 — APPROVED** [review]

**Review by JillieBeanSim — APPROVED** [review]
Thanks @t1m0thyj this LGTM!

**Review by traeok — APPROVED** [review]

**Pull request merged**